### PR TITLE
eslint errors

### PR DESCRIPTION
**Pull request by dnafication**
Link #1461 (fix eslint warnings)

- `max-len`: most of the sensible `max-len` errors are fixed
- `no-unused-vars`: most of these types of errors are fixed by deleting not needed declarations
- `@typescript-eslint/no-unused-vars`: same as above

Tests pass locally.

### CODE REVIEW

**Comment by dnafication** [comment]
🤔 why are the tests failing in CI environment? My local env was 

- `node v14.11.0 `
- `pnpm 5.10.1`

**Comment by juanpicado** [comment]
> 🤔 why are the tests failing in CI environment? My local env was
> 
> * `node v14.11.0 `
> * `pnpm 5.10.1`

hard to know, you might go one rule at a time.

File: packages/auth/test/auth-utils.spec.ts [hunk 1]
```diff
@@ -428,7 +427,7 @@ describe('Auth utilities', () => {
         );
       });
 
-      test('should fail on verify the token and return anonymous users', async () => {
+      test('should fail on verify the token and return anonymous users 2', async () => {
```
Comment (juanpicado) [inline packages/auth/test/auth-utils.spec.ts:430]: ??

Comment (dnafication) [reply packages/auth/test/auth-utils.spec.ts:430]: Eslint complains about tests with same name. There were two tests with same name in this file.

Comment (juanpicado) [reply packages/auth/test/auth-utils.spec.ts:430]: Let's reword this one, actually the test does `should verify the token and return a remote user`. I think that's the right description here.

File: packages/middleware/src/middleware.ts [hunk 2]
```diff
@@ -195,12 +197,7 @@ export interface MiddlewareError {
 
 export type FinalBody = Package | MiddlewareError | string;
 
-export function final(
-  body: FinalBody,
-  req: $RequestExtend,
-  res: $ResponseExtend,
-  next: $NextFunctionVer
```
Comment (juanpicado) [inline packages/middleware/src/middleware.ts:202]: You are removing `next`, that explain why all test are failing. This is being used in that method.

**Review by juanpicado — CHANGES_REQUESTED** [review]
Check my comment :+1:

File: packages/auth/test/auth-utils.spec.ts (hunk 1 above)
Comment (dnafication) [reply packages/auth/test/auth-utils.spec.ts:430]: Sure. I was't sure what should go there. Thanks.

File: packages/middleware/src/middleware.ts (hunk 2 above)
Comment (dnafication) [reply packages/middleware/src/middleware.ts:202]: Oh... eslint was complaining this as unused var. I will revert this change.

**Comment by dnafication** [comment]
Tests seems to pass now but Code scanning/CodeQL has errors. May be we should look at fixing them in other issues.

**Review by juanpicado — APPROVED** [review]
Great stuff 🙏❤️ !!

**Comment by juanpicado** [comment]
> Tests seems to pass now but [Code scanning/CodeQL](https://github.com/verdaccio/verdaccio/pull/1973/checks?check_run_id=1300706284) has errors. May be we should look at fixing them in other issues. 

It is fine, we have enabled them only for master, the idea is use that info to improve code in 5.x eventually, but for now still to early. Thanks for pointing this out.

**Comment by juanpicado** [comment]
Maybe. Before merge you want to use the changeset feature I am testing, check the contributions the contributions guideline.

**Comment by dnafication** [comment]
Hi @juanpicado 
Just to confirm, you want me to run `pnpm changeset` on the branch?

**Comment by juanpicado** [comment]
> Hi @juanpicado
> Just to confirm, you want me to run `pnpm changeset` on the branch?

yes, you run `pnpm changeset` , follow the instructions, add this changes as a **patch**, add your fancy comment, all that info will be visible in the changelog, you can tag yourself if you like, then commit and push, then I will merge.

**Comment by juanpicado** [comment]
From 198 warnings to 78 :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap:

**Pull request merged**